### PR TITLE
fix(agents): restore model context size and capability icons in the model picker

### PR DESCRIPTION
**Pull request by Maxsky5**
## Problem

The reworked model picker dropped useful per-model metadata: the max context window was no longer shown, the second row repeated the runtime name, and there was no quick way to see which models accept images, videos, audio, or PDF attachments.

## Goal

Restore the context window label (e.g. "200K context") when the runtime reports it, drop the redundant runtime name from the second row, and add small capability icons next to the context size with a one-line tooltip per capability.

## Changes

- Model rows show `Provider · modelId` on the second line; capability icons and the context window are right-aligned and vertically centered so they line up with the favorite button.
- Each capability icon explains what the model accepts in a tooltip.
- The selected model is marked with an accent bar on the row's left edge plus the accent background — pure paint, so the model name, provider, and context size never shift when selection changes.
- Context and capability metadata render only when the runtime catalog provides them (OpenCode does; the Codex and Claude SDKs do not expose a context window today, so those rows intentionally omit it).
- Shared tooltip fixes: tooltip content now paints above dialogs and refreshes when moving between adjacent triggers.

## Summary by CodeRabbit

* **New Features**
  * Model picker rows now display context-window sizes and supported image, video, and PDF attachments.
  * Selected models clearly highlight their metadata and indicators.
  * Token counts use compact or exact numeric formats, including improved million-value rounding.

* **Bug Fixes**
  * Improved tooltip layering and hover behavior for more reliable capability details.
  * Corrected model metadata descriptions when capabilities or context details are unavailable.

### CODE REVIEW

File: packages/frontend/src/components/features/agents/model-picker/model-picker.tsx [hunk 1]
```diff
@@ -177,6 +187,61 @@ const RuntimeRailButton = ({
...
+  if (capabilities.length === 0) {
+    return null;
+  }
+  return (
+    <span className="flex shrink-0 items-center gap-1">
+      {capabilities.map(({ icon: Icon, label, description }) => (
+        <Tooltip key={label}>
+          <TooltipTrigger asChild>
+            <span role="img" aria-label={label}>
+              <Icon className="size-3.5" aria-hidden="true" />
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/components/features/agents/model-picker/model-picker.tsx:235]: Fixed in 278781ef2. The row button now carries an `aria-description` that includes the context window and supported capabilities (e.g. `200K token context window. Supports images, videos, pdf files`), so screen readers get the metadata without the icon spans needing to be focusable. The `aria-description` is also asserted in the new accent-bar test.

File: packages/frontend/src/components/ui/tooltip.tsx [hunk 2]
```diff
@@ -18,8 +18,17 @@ function TooltipProvider({
   );
 }
 
-function Tooltip({ ...props }: React.ComponentProps<typeof TooltipPrimitive.Root>) {
-  return <TooltipPrimitive.Root data-slot="tooltip" {...props} />;
+function Tooltip({
+  disableHoverableContent = true,
+  ...props
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/components/ui/tooltip.tsx]: Fixed in 278781ef2. Reverted the shared `Tooltip` wrapper back to Radix's default hoverable behavior and scoped `disableHoverableContent` to the capability-icon tooltips only, where the adjacent-trigger staleness was actually user-visible. The other ~37 dependents (including the kanban-card tooltip) keep hoverable content.

File: packages/frontend/src/components/features/agents/model-picker/model-picker.tsx [hunk 3]
```diff
@@ -247,14 +313,25 @@ const ModelRow = ({
...
+            {item.model.providerName} · {item.model.modelId}
           </span>
         </span>
-        {selected ? <Check aria-label="Selected model" /> : null}
+        <span className="ml-auto flex shrink-0 items-center gap-2 self-center text-xs text-muted-foreground">
+          <ModelCapabilityIcons support={item.model.attachmentSupport} />
+          {contextWindowLabel ? (
+            <span className="shrink-0">{contextWindowLabel} context</span>
+          ) : null}
+        </span>
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/components/features/agents/model-picker/model-picker.tsx]: Thanks for the feedback. We're intentionally keeping the context label right-anchored and non-shrinking: a stable context anchor was an explicit product requirement for this picker, and the capability icons plus context must line up with the favorite action regardless of row content. The model name/provider column already truncates with an ellipsis to adapt to narrow widths, which is the designed behavior here, and the picker is capped at 42rem. Not changing this.

**Comment by Maxsky5** [comment]
Thanks for the detailed review. Handling of each item:

Worth checking
- Selection accent / no-reflow test: added ("marks the selected row with a paint-only accent bar") — asserts the accent span exists only on the selected row and the row button's accessible description.
- Tooltip coverage: the tooltip test now also hovers the video icon, and two partial-metadata fixtures were added (context without attachment support, and icons without a context window).

Small things
- Selected-row metadata contrast: the provider line and the right-aligned metadata cluster now use text-accent-foreground on the selected row.
- Tooltip z-index: kept z-[80] (above the dialog's z-[70]); the arrow's stray z-50 was removed so content and arrow share the same layer. A z-index regression can't be asserted in jsdom, and the collision with toasts/update-prompt is an accepted DOM-order edge case.
- disableHoverableContent: reverted the shared default and scoped it to the capability-icon tooltips; the adjacent-trigger refresh test lives in the model picker where the behavior now lives.

**Comment by Maxsky5** [comment]
Both actionable findings are fixed in 278781ef2:
- format-token-count: 999.5K and above now promote to 1M (with unit tests).
- model-picker tests: the runtimes fixture is now a per-test factory returning fresh nested objects.

The pre-merge checklist warnings (PR description template sections and 80% docstring coverage) are CodeRabbit template defaults, not requirements of this repo — the description already covers problem, goal, and changes, and the repo has no docstring-coverage policy. No action taken.

**Comment by Maxsky5** [comment]
The responsive-layout comment was answered in its thread: the right-anchored context label is an explicit product requirement and the name/provider column truncates to adapt. No other open items from the Greptile summary.
